utils/navigator_gui/navigator_gui/dashboard.py

```diff
@@ -0,0 +1,486 @@
+        self.battery_critical_voltage = rospy.get_param("/battery_monitor/battery_critical_voltage", 20.6)
+
+        rospy.Subscriber("/battery_monitor", Float32, self.cache_battery_voltage)
+        rospy.Subscriber("/clock", Clock, self.cache_system_time)
+
+        self.wrench_changer = rospy.ServiceProxy('/change_wrench', WrenchSelect)
+        self.station_holder = rospy.ServiceProxy('/lqrrt/station_hold', SetBool)
+
+        self.kb = KillBroadcaster(id='station_hold', description='Resets Pose')
+        self.kill_listener = AlarmListener('kill', self.update_kill_status)
+
+        alarm_broadcaster = AlarmBroadcaster()
+        self.kill_alarm = alarm_broadcaster.add_alarm(
+            name='kill',
+            action_required=True,
+            severity=0
+        )
+
+    def update_kill_status(self, alarm):
+        '''
+        Updates the kill status display when there is an update on the kill
+        alarm. Caches the last displayed kill status to avoid updating the
+        display with the same information twice.
+        '''
+        if (not self.system_time["is_timed_out"]):
+
+            if (alarm.clear):
+                if (self.is_killed):
+                    self.is_killed = False
+                    self.kill_status_status.setText("Alive")
+                    self.kill_status_frame.setStyleSheet(self.colors["green"])
+
+            elif (not self.is_killed):
+                self.is_killed = True
+                self.kill_status_status.setText("Killed")
+                self.kill_status_frame.setStyleSheet(self.colors["red"])
+
+    def monitor_operating_mode(self):
+        '''
+        Monitors the selected wrench input device on a 0.5s interval using the
+        wrench_changer service. Only updates the display when the selected
+        wrench input has changed.
+        '''
+        if (not self.system_time["is_timed_out"]):
+
+            # Attempting to set the wrench to an empty string fails and returns the selected input device.
+            try:
+                self.operating_mode["received"] = self.wrench_changer("").str
+
+            # If calling the service fails, the operating mode is set to unknown
+            except:
+                self.operating_mode["received"] = "Unknown"
+
+            # If a new value was received, update the display
+            if (self.operating_mode["received"] != self.operating_mode["cached"]):
+                self.update_operating_mode_status()
+
+        # Schedules the next instance of this method with a QT timer
+        QtCore.QTimer.singleShot(500, self.monitor_operating_mode)
+
+    def update_operating_mode_status(self):
+        '''
+        Updates the displayed operating mode status text and color.
+        '''
+        if (self.operating_mode["received"] == "autonomous"):
+            self.operating_mode_status.setText("Autonomous")
+            self.operating_mode_frame.setStyleSheet(self.colors["green"])
```

Comment:
The colors are simply there to make the display glanceable. My thinking was that It is far easier to look at it and see a color for each item on the display than to read four words or values.